Support building promtail without systemd/journald support (disabling cgo)

**Describe the bug**
Several linux distributions do not ship with systemd. Building promtail without journald scraping support (which is useless on those distributions anyway) would be useful.

**To Reproduce**
Steps to reproduce the behavior:
0. run an Alpine linux container (e.g. `podman run -ti  --rm library/alpine`)
1. get the latest loki release
2. run `make promtail`
3. see the following error:
```
CGO_ENABLED=1 go build -ldflags "-s -w -X github.com/grafana/loki/pkg/build.Branch= -X github.com/grafana/loki/pkg/build.Version= -X github.com/grafana/loki/pkg/build.Revision= -X github.com/grafana/loki/pkg/build.BuildUser=root@f71acacd48c9 -X github.com/grafana/loki/pkg/build.BuildDate=2021-07-02T08:20:03Z" -tags netgo -mod=vendor -o cmd/promtail/promtail ./cmd/promtail
# github.com/coreos/go-systemd/sdjournal
vendor/github.com/coreos/go-systemd/sdjournal/journal.go:27:11: fatal error: systemd/sd-journal.h: No such file or directory
   27 | // #include <systemd/sd-journal.h>
      |           ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:222: cmd/promtail/promtail] Error 2
```

It seems that the Makefile *unconditionally* sets `PROMTAIL_CGO` when the system is linux. There should be a way for the user compiling promtail to override this behaviour in the Makefile.
```
# Validate GOHOSTOS=linux && GOOS=linux to use CGO.
ifeq ($(shell go env GOHOSTOS),linux)
ifeq ($(shell go env GOOS),linux)
PROMTAIL_CGO = 1
PROMTAIL_GO_FLAGS = $(DYN_GO_FLAGS)
PROMTAIL_DEBUG_GO_FLAGS = $(DYN_DEBUG_GO_FLAGS)
endif
endif
```

**Expected behavior**
Some way to *disable* PROMTAIL_CGO when compiling on linux.

--- a/Makefile
+++ b/Makefile
@@ -202,13 +202,13 @@ PROMTAIL_GO_FLAGS := $(GO_FLAGS)
 PROMTAIL_DEBUG_GO_FLAGS := $(DEBUG_GO_FLAGS)
 
 # Validate GOHOSTOS=linux && GOOS=linux to use CGO.
-ifneq ($(CGO_ENABLED), 0)
 ifeq ($(shell go env GOHOSTOS),linux)
 ifeq ($(shell go env GOOS),linux)
+ifneq ($(CGO_ENABLED), 0)
 PROMTAIL_CGO = 1
+endif
 PROMTAIL_GO_FLAGS = $(DYN_GO_FLAGS)
 PROMTAIL_DEBUG_GO_FLAGS = $(DYN_DEBUG_GO_FLAGS)
-endif
 endif
 endif
 

--- a/Makefile
+++ b/Makefile
@@ -202,11 +202,13 @@ PROMTAIL_GO_FLAGS := $(GO_FLAGS)
 PROMTAIL_DEBUG_GO_FLAGS := $(DEBUG_GO_FLAGS)
 
 # Validate GOHOSTOS=linux && GOOS=linux to use CGO.
+ifneq ($(CGO_ENABLED), 0)
 ifeq ($(shell go env GOHOSTOS),linux)
 ifeq ($(shell go env GOOS),linux)
 PROMTAIL_CGO = 1
 PROMTAIL_GO_FLAGS = $(DYN_GO_FLAGS)
 PROMTAIL_DEBUG_GO_FLAGS = $(DYN_DEBUG_GO_FLAGS)
+endif
 endif
 endif
 

